src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -73,6 +83,39 @@ public int getReplacementOperationsTotalDifference() {
         return replacementOperationsTotalDifference;
     }
 
+
+    /**
+     * Expands current json path with the key reference
+     */
+    public void expandCurrentJsonPath(int keyStartIndex, int keyLength) {
+        currentJsonPath.push(new int[]{keyStartIndex, keyLength});
+    }
+
+    /**
+     * Backtracks to the previous json path component
+     */
+    public void backtrackCurrentJsonPath() {
+        currentJsonPath.pop();
+    }
+
+    /**
+     * Checks if the masker is in json array node
+     */
+    public boolean isInArray() {
+        return !currentJsonPath.isEmpty() && currentJsonPath.peek()[1] == -1;
+    }
+
+    public void incrementCurrentJsonPathArrayIndex() {
+        currentJsonPath.peek()[0]++;
```

Comment:
I don't know why, but this line of code looks so odd 😂

Reply:
It is though. I will think about how to make the reference representation comprehensible.

Reply:
Done.